FileDialog wildcard documentation fix and workaround note for macOS

julbean opened the pull request:
FileDialog wildcard documentation fix and bug note with workaround

julbean committed:
Update filedlg.h

FileDialog documentation fix and bug note with workaround

julbean committed:
Update filedlg.h

Removed trailing white space

swt2c commented on interface/wx/filedlg.h:
```diff
@@ -112,12 +112,15 @@ const char wxFileSelectorDefaultWildcardStr[];
     the file open dialog:
     @code
-         wxSystemOptions::SetOption(wxOSX_FILEDIALOG_ALWAYS_SHOW_TYPES, 1)
+         wxSystemOptions::SetOption('osx.openfiledialog.always-show-types', 1)
```
I think the proper fix here (for the downstream wxPython issue [https://github.com/wxWidgets/Phoenix/issues/2368]) is that the `#define wxOSX_FILEDIALOG_ALWAYS_SHOW_TYPES` from `include/wx/osx/filedlg.h` should be replicated in this file.  That would make the constant available to wxPython.

julbean replied:
Sounds good, I'm just a newbie when it comes to C code so didn't know about the better way.  Changed it to use the #define

vadz commented on interface/wx/filedlg.h:
```diff
@@ -112,12 +112,15 @@ const char wxFileSelectorDefaultWildcardStr[];
+    file type are greyed out and are not selectable. Also, currently the Mac macOS
+    implementation is bugged, causing all filters after the first to also have the
+    first filter applied, so ensure your first filter is a superset of all other
+    wildcard filters to avoid any issues.
```
Sorry, "is bugged" is just not the right way to say it. If I understand the issue correctly, I would rather say something like

```
@note Due to a macOS limitation, the first filter should be a catch-all one,
      i.e. use `*` as the wildcard, as it is always applied in addition to
      all the other filters, when another one is selected.
```

Of course, if it's possible to fix this, it would be even better. @csomor Do you know if this is really a macOS limitation or just wxOSX bug?

julbean committed:
Update filedlg.h

Reverted SetOption changed, replaced with copying #define from https://github.com/wxWidgets/wxWidgets/blob/master/include/wx/osx/filedlg.h

julbean committed:
Revert workaround note, fix small typo

Although it might be nice to note the workaround for prior versions which have this bug, due to the presumable rarity of its circumstances (unnoticed until now) it seems unnecessary to clutter the documentation with it, as new projects should be using the newer fixed releases and older projects requiring older bugged versions are unlikely to add a feature that triggers the bug.
Also fixed typo: does -> do

julbean commented:
Reverted the workaround note, as csomor has fixed the bug for future versions (although I haven't been able to test it to confirm).
Now this PR just fixes the ALWAYS_SHOW_TYPES issue (https://github.com/wxWidgets/Phoenix/issues/2368) and now also fixes a small typo.

vadz commented on interface/wx/filedlg.h:
```diff
@@ -20,6 +20,11 @@ enum
 
 #define wxFD_DEFAULT_STYLE      wxFD_OPEN
 
+// set this system option to 1 in order to always show the filetypes popup in
```
Does this actually work, i.e. does this expose this symbol in wxPython? Or do we need to use `///` or `/**` comments for it to be picked up?

julbean committed:
Revert copied irrelevant comment

Reverted copying irrelevant comment from include/wx/osx/filedlg.h
